### PR TITLE
[BMJAS-2631] Fix unregisterListeners which did not correct removeEventListener

### PR DESCRIPTION
**Pull request by gingman**
`.bind(this)` creates a copy of the method so the `removeEventListener` was not made on the correct reference so it never really detached.

This bug was in the original fork and also in their `video-jwplayer-js`. Probably that no one uses the pattern that we do.

### CODE REVIEW

**Review by alarochelle — APPROVED** [review]
Should we also open a PR in the official repo?

File: src/tracker.js [hunk 1]
```diff
@@ -69,35 +69,49 @@ export default class ShakaTracker extends nrvideo.VideoTracker {
...
-    this.tag.removeEventListener('playing', this.onPlaying)
-    this.tag.removeEventListener('pause', this.onPause)
-    this.tag.removeEventListener('seeking', this.onSeeking)
-    this.tag.removeEventListener('seeked', this.onSeeked)
-    this.tag.removeEventListener('error', this.onError)
-    this.tag.removeEventListener('ended', this.onEnded)
-
-    this.player.removeEventListener('buffering', this.onBuffering)
-    this.player.removeEventListener('adaptation', this.onAdaptation)
+    console.log("Unregistering all events for my Video Tracker")
```
Comment (Boubalou) [inline src/tracker.js]:
```suggestion
```

**Review by Boubalou — APPROVED** [review]
🎉

**Comment by gingman** [comment]
@alarochelle Repo hasn't been updated for 5 years. I don't think they care 😄

**Comment by alarochelle** [comment]
> @alarochelle Repo hasn't been updated for 5 years. I don't think they care 😄

It was updated last year actually. I contributed "recently" to the iOS repo and they merged my PR.. I think it's worth it.

**Comment by gingman** [comment]
I would have to fork another repo. This one also has changes for Shaka 3.x

**Pull request merged**